src/app/payment/paymentMethod/CheckoutcomCustomFormFields.spec.tsx

```diff
@@ -175,24 +193,57 @@ describe('CheckoutCustomFormFields', () => {
 
     describe('Sepa form fieldset', () => {
         const SepaFormFieldset = checkoutcomCustomFormFields.sepa;
+        let component: ReactWrapper;
+        const sepaProps = getAPMProps.sepa();
+
+        beforeEach(() => {
+            component = mount(<CheckoutcomAPMsTest { ...sepaProps } cardFieldset={ <SepaFormFieldset debtor={ getBillingAddress() } method={ sepaProps.method } /> } />);
+        });
 
         it('should render the sepa fieldset', () => {
-            const sepaProps = getAPMProps.sepa();
-            const compoonent = mount(<CheckoutcomAPMsTest { ...sepaProps } cardFieldset={ <SepaFormFieldset debtor={ getBillingAddress() } method={ sepaProps.method } /> } />);
+            expect(component.find('input[name="iban"]')).toHaveLength(1);
+            expect(component.find('input[name="sepaMandate"]')).toHaveLength(1);
+        });
+
+        it('should call toggleSubmitButton on checkbox checked', () => {
+            const checkbox = component.find('input[name="sepaMandate"]');
+            checkbox.simulate('change', {target: {name: 'sepaMandate', value: true}});
 
-            expect(compoonent.find('input[name="iban"]')).toHaveLength(1);
-            expect(compoonent.find('input[name="sepaMandate"]')).toHaveLength(1);
+            expect(paymentContext.disableSubmit).lastCalledWith(sepaProps.method, false);
+        });
+
+        it('should call disableSubmit on useEffect cleanup function', () => {
+            component.unmount();
+
+            expect(paymentContext.disableSubmit).lastCalledWith(sepaProps.method, false);
+        });
+
+        afterEach(() => {
+            jest.clearAllMocks();
         });
     });
 
     describe('iDeal form fieldset', () => {
         const IdealFormFieldset = checkoutcomCustomFormFields.ideal;
+        let component: ReactWrapper;
 
-        it('should render the ideal fieldset', () => {
+        beforeEach(() => {
             const idealProps = getAPMProps.ideal();
-            const compoonent = mount(<CheckoutcomAPMsTest { ...idealProps } cardFieldset={ <IdealFormFieldset method={ idealProps.method } /> } />);
+            component = mount(<CheckoutcomAPMsTest { ...idealProps } cardFieldset={ <IdealFormFieldset method={ idealProps.method } /> } />);
+        });
+
+        it('should render the ideal fieldset', () => {
 
-            expect(compoonent.find('input[type="hidden"]')).toHaveLength(1);
+            expect(component.find('input[type="hidden"]')).toHaveLength(1);
+        });
+
+        it('should call handleclick on DropdownTrigger change', () => {
```

Comment:
Can we please add a description of what  is happening on the UI (i.e. Shopper selects a payment method from dropdown, etc.)